src/ui-setup/scripts/generate-ui.js

```diff
@@ -1,34 +1,48 @@
-const ncp = require('ncp')
 const fs = require('fs')
+const fse = require('fs-extra')
 const chalk = require('chalk')
 const path = require('path')
+const { exec } = require('child_process')
 const renderSchema = require('./renderSchema')
 const renderResolvers = require('./renderResolvers')
 const renderHomePage = require('./renderHomePage')
 const renderTypePage = require('./renderTypePage')
+const renderTypePageTest = require('./renderTypePageTest')
+const renderTypePageIntegrationTest = require('./renderTypePageIntegrationTest')
 const renderIndex = require('./renderIndex')
 const mapObject = require('./render-utils').mapObject
 const typeSpecPath = process.argv[2]
-const defaultTypeSpecPath = path.resolve("src/setup/type-specs", "sample-type-spec.json");
+const defaultTypeSpecPath = path.resolve('src/setup/type-specs', 'sample-type-spec.json')
 const defaultTypeSpec = require(defaultTypeSpecPath)
+const { capitalize, toCamelCase } = require('../../setup/utils.js')
+
+// map over type spec to ensure the types are all capitalized before feeding the name to the UI generation
+const formCSTypeSpec = typeSpec => {
+  const rawTypeSpecTypesMap = mapObject(typeSpec.types, (typeName, type) => [capitalize(toCamelCase(typeName)), type])
+  const rawTypeSpec = { types: Object.fromEntries(rawTypeSpecTypesMap) }
+  return rawTypeSpec
+}
 
 let typeSpec
 if (!typeSpecPath) {
-  console.log(chalk.blue('> No type spec JSON file provided. \n  Using default type spec JSON file located within the setup directory.\n'))
-  typeSpec = defaultTypeSpec
-}
-else {
-  typespec = JSON.parse(fs.readFileSync(typeSpecPath))
+  console.log(chalk.blue('> No type spec JSON file provided. \n Using default type spec JSON file located within the setup directory.\n'))
+  const updatedTypeSpec = formCSTypeSpec(defaultTypeSpec)
+  typeSpec = updatedTypeSpec
+} else {
+  const rawTypeSpec = JSON.parse(fs.readFileSync(typeSpecPath))
+  const updatedTypeSpec = formCSTypeSpec(rawTypeSpec)
+  typeSpec = updatedTypeSpec
```

Comment:
this (together with lines 26, 27 and 35 (awkward github)) can be simplified to

```
const rawTypeSpec = !typeSpecPath
  ? defaultTypeSpec
  :  JSON.parse(fs.readFileSync(typeSpecPath))
const typeSpec = capitalizeTypeNames(rawTypeSpec)

Reply:
Agreed - I was originally keeping this logic in the if/then format to provide a console about referencing the default, but I'll just separate that part from the remaining logic.